addons/mail/static/src/models/message_action_list/message_action_list.js

```diff
@@ -44,6 +44,7 @@ function factory(dependencies) {
                 body: '',
                 attachment_ids: [],
             });
+            this.message.toggleStar();
```

Comment:
It's no good to perform toggle star RPC for all messages that are being deleted, whether starred or not.
Also this LOC is off. It's strange to see concern of starred in the delete flow.

Ideally, starred counter should have awareness of which messages are starred, and when a message is considered deleted it should automatically change the counter.

Right now the client-side code has no such awareness and instead fully trust what the server tells for how to change the counter. So the proper fix with this design should have been for the server-side code to notice the message being "deleted" was starred, and therefore notify client-side agent starred counter should be updated.

I'm not sure how easy/hard it is to make this fix in python code as I don't have enough experience in the python code of discuss...

Reply:
@alexkuhn Thanks for your feedback.
We will look forward to your suggestions at python side.

Reply:
@alexkuhn As you suggested I have updated my code and put the logic in server side so now on wards it should automatically change the counter, we don't need to call method explicitly.